content/4_Gold/Modular.mdx

```diff
@@ -181,42 +181,45 @@ public class Exponentation
 
 ## Modular Inverse
 
+<FocusProblem problem="binomial" />
+
+<Resources>
+	<Resource
+		source="cp-algo"
+		title="Modular Multiplicative Inverse"
+		url="algebra/module-inverse.html"
+		starred
+	>
+		Various ways to take modular inverse, we'll only discuss the second here
+	</Resource>
+</Resources>
+
 The **modular inverse** is the equivalent of the reciprocal in real-number
 arithmetic; to divide $a$ by $b$, multiply $a$ by the modular inverse of $b$.
 We'll only consider prime moduli $p$ here.
 
-For example, the inverse of $2$ modulo $p=10^9+7$ is
-$i=\frac{p+1}{2}=5\cdot 10^8+4$. This means that for any integer $x$,
-
+For example, the inverse of $2 \pmod{p}$ where $p=10^9+7$ is
 $$
-(2x)\cdot i\equiv x\cdot (2i)\equiv x\pmod{10^9+7}.
+1\equiv p+1 \pmod{p}.
 $$
 
-For example, $10i\equiv 5\pmod{10^9+7}$.
-
-<!-- #### Example: Sprinkers 2
+$$
+2\cdot i\equiv p+1 \pmod{p}.
+$$
```

Comment:
$i$ kinda comes out of nowhere. Could be beneficial to either explain it beforehand, or just use $2^{-1}$ as notation?